src/Microsoft.ML.Transforms/Text/WordEmbeddingsTransform.cs

```diff
@@ -0,0 +1,465 @@
+            ch.Assert(0 <= iinfo && iinfo < Infos.Length);
+
+            var info = Infos[iinfo];
+            ch.Assert(info.TypeSrc.IsVector);
+            ch.Assert(info.TypeSrc.ItemType.IsText);
+
+            var srcGetter = input.GetGetter<VBuffer<DvText>>(info.Source);
+            var src = default(VBuffer<DvText>);
+            float[] wordVector = new float[_currentVocab.Dimension];
+
+            return
+                (ref VBuffer<float> dst) =>
+                {
+                    int deno = 0;
+                    srcGetter(ref src);
+                    var values = dst.Values;
+                    Utils.EnsureSize(ref values, 3 * _dimension, keepOld: false);
+                    Array.Clear(values, 0, 3 * _dimension);
+                    for (int i = 0; i < src.Count; i++)
+                        deno = FillOutputVector(ref src.Values[i], values, wordVector, deno);
+
+                    if (deno != 0)
+                        for (int index = 0; index < _dimension; index++)
+                            values[index + _dimension] /= deno;
+
+                    dst = new VBuffer<float>(values.Length, values, dst.Indices);
+                };
+        }
+
+        private int FillOutputVector(ref DvText src, float[] val, float[] wordVector, int deno)
+        {
+            if (_currentVocab.GetWordVector(ref src, wordVector))
+            {
+                deno++;
+                int offset = 2 * _dimension;
+                for (int i = 0; i < _dimension; i++)
+                {
+                    float currentTerm = wordVector[i];
+                    val[i] = val[i] < currentTerm ? val[i] : currentTerm;
+                    val[_dimension + i] += currentTerm;
+                    val[offset + i] = val[offset + i] > currentTerm ? val[offset + i] : currentTerm;
+                }
+            }
+            return deno;
+        }
+
+        public enum PretrainedModelKind
+        {
+            [TGUI(Label = "GloVe 50D")]
+            GloVe50D = 0,
+
+            [TGUI(Label = "GloVe 100D")]
+            GloVe100D = 1,
+
+            [TGUI(Label = "GloVe 200D")]
+            GloVe200D = 2,
+
+            [TGUI(Label = "GloVe 300D")]
+            GloVe300D = 3,
+
+            [TGUI(Label = "GloVe Twitter 25D")]
+            GloVeTwitter25D = 4,
+
+            [TGUI(Label = "GloVe Twitter 50D")]
+            GloVeTwitter50D = 5,
+
+            [TGUI(Label = "GloVe Twitter 100D")]
+            GloVeTwitter100D = 6,
+
+            [TGUI(Label = "GloVe Twitter 200D")]
+            GloVeTwitter200D = 7,
+
+            [TGUI(Label = "fastText Wikipedia 300D")]
+            FastTextWikipedia300D = 8,
+
+            [TGUI(Label = "Sentiment-Specific Word Embedding")]
+            Sswe = 9
+        }
+
+        private static Dictionary<PretrainedModelKind, string> _modelsMetaData = new Dictionary<PretrainedModelKind, string>()
+        {
+             { PretrainedModelKind.GloVe50D, "glove.6B.50d.txt" },
```

Comment:
>GloVe50D [](start = 35, length = 8)

do we have public locations for all of this? 
Do we want to be the ones storing them?

Reply:
we have aka.ms/tlc-resources/

---
In reply to: [204168101](https://github.com/dotnet/machinelearning/pull/545#discussion_r204168101) [](ancestors = 204168101)